src/Select/Select.js

```diff
@@ -0,0 +1,234 @@
+    border,
+    boxShadow: 'none',
+    ':hover': {
+      border: isError
+        ? '1px solid var(--persimmon-hover)'
+        : '1px solid var(--jungle-green)'
+    }
+  }
+}
+
+const inputStyles = provided => ({ ...provided, margin: 0, padding: 0 })
+
+const menuStyles = provided => ({
+  ...provided,
+  top: 'unset',
+  background: 'var(--white)',
+  boxShadow: '0px 4px 8px rgba(24, 27, 43, 0.03)',
+  border: '1px solid var(--porcelain)',
+  boxSizing: 'border-box',
+  zIndex: 2,
+  minHeight: 48
+})
+
+const valueContainerStyles = (provided, state) => ({
+  ...provided,
+  padding: state.isMulti ? 4 : '0 10px',
+  boxSizing: 'border-box',
+  gap: 4
+})
+
+const singleValueStyles = (provided, state) => ({
+  ...provided,
+  color: state.isDisabled ? 'var(--mystic)' : 'var(--rhino)'
+})
+
+const multiValueStyles = provided => ({
+  ...provided,
+  display: 'flex',
+  alignItems: 'center',
+  height: 22,
+  margin: 0,
+  padding: '0 8px',
+  background: 'var(--white)',
+  border: '1px solid var(--porcelain)',
+  borderRadius: 4
+})
+
+const multiValueLabelStyles = (provided, state) => ({
+  ...provided,
+  padding: 0,
+  paddingLeft: 0,
+  color: state.isDisabled ? 'var(--mystic)' : 'var(--rhino)'
+})
+
+const multiValueRemoveStyles = provided => ({
+  ...provided,
+  padding: 0,
+  marginLeft: 6,
+  fill: 'var(--waterloo)'
+})
+
+const placeholderStyles = (provided, state) => ({
+  ...provided,
+  color: state.isDisabled ? 'var(--mystic)' : 'var(--casper)'
+})
+
+const dropdownIndicatorStyles = (provided, state) => ({
+  ...provided,
+  transform: state.selectProps.menuIsOpen && 'rotate(180deg)',
+  padding: '10px 12px',
+  fill: state.isDisabled ? 'var(--mystic)' : 'var(--waterloo)',
+  cursor: 'pointer'
+})
+
+const clearIndicatorStyles = (provided, state) => ({
+  ...provided,
+  padding: '0 12px',
+  fill: state.isDisabled ? 'var(--mystic)' : 'var(--waterloo)',
+  cursor: 'pointer'
+})
+
+const indicatorSeparatorStyles = provided => ({
+  ...provided,
+  alignSelf: 'stretch',
+  backgroundColor: 'var(--porcelain)',
+  marginBottom: 0,
+  marginTop: 0,
+  width: 1,
+  minHeight: 30
+})
+
+const indicatorContainerStyles = provided => ({
+  ...provided,
+  height: 30
+})
+
+const indicatorsContainerStyles = provided => ({
+  ...provided,
+  justifyContent: 'center'
+})
+
+const getStyles = ({ isError, customStyles }) => ({
+  container: containerStyles,
+  control: controlStyles(isError),
+  input: inputStyles,
+  menu: menuStyles,
+  valueContainer: valueContainerStyles,
+  singleValue: singleValueStyles,
+  multiValue: multiValueStyles,
+  multiValueLabel: multiValueLabelStyles,
+  multiValueRemove: multiValueRemoveStyles,
+  placeholder: placeholderStyles,
+  dropdownIndicator: dropdownIndicatorStyles,
+  clearIndicator: clearIndicatorStyles,
+  indicatorSeparator: indicatorSeparatorStyles,
+  indicatorContainer: indicatorContainerStyles,
+  indicatorsContainer: indicatorsContainerStyles,
+  ...customStyles
+})
```

Comment:
This approach has 2 disadvantages:
- Creating a big object on every render (slower than css);
- Bloats the JS file (Can't be downloaded in parallel as a css resource)

Better to move this styles to scss module and assign classes as you have done on `LOC 42` and `LOC 48`.

State dependent styles (e.g. `state.isDisabled ? 'var(--mystic)' : 'var(--waterloo)',`) can be implemented using enum-like classes and css variables